### PR TITLE
fix: certifications Terraform exam objectives content

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎

## 🗒️ What

Updates the exam FAQ content for `Terraform Associate 002` and `Terraform  Associate 003`.

## 📸 Design Screenshots

| Exam | Before | After |
| - | - | - |
| Terraform Associate 002 | 002-before | 002-after |
| Terraform Associate 003 | 003-before | 003-after |

## 🧪 Testing

- [ ] Visit [/certifications/infrastructure-automation][/certifications/infrastructure-automation]
    - The exam FAQ content for `Terraform Associate 002` should be match the content in the table shown on <https://developer.hashicorp.com/terraform/tutorials/certification/associate-review>
    - The exam FAQ content for `Terraform Associate 003` should be match the content in the table shown on <https://developer.hashicorp.com/terraform/tutorials/certification-003/associate-review-003>

[preview]: https://dev-portal-git-zscert-tf002-content-hashicorp.vercel.app/certifications/infrastructure-automation
[/certifications/infrastructure-automation]: https://dev-portal-git-zscert-tf002-content-hashicorp.vercel.app/certifications/infrastructure-automation

### CODE REVIEW

**Comment by zchsh** [comment]
@lszpunar @emilypersson1 This PR updates the `Exam Objectives` content:
- Updates Terraform Associate 002 to match the content on https://developer.hashicorp.com/terraform/tutorials/certification/associate-review
- Updates Terraform Associate 003 to match the content on https://developer.hashicorp.com/terraform/tutorials/certification-003/associate-review-003

I've left things with links & all since that was what was quickest to do. Happy to help modify the content if there are any changes we want to make in this context 🙇

**Review by emilypersson1 — APPROVED** [review]
looks good to me, thank you for the quick fix @zchsh ! also sent this to @lszpunar in slack for her review

File: src/content/certifications/exam-faqs/terraform-associate-002.mdx [hunk 1]
```diff
@@ -18,70 +18,61 @@
...
+| **2** | **Understand Terraform's purpose (vs other IaC)**                                                                   |
+| 2a    | Explain multi-cloud and provider-agnostic benefits                                                                  |
+| 2b    | Explain the benefits of state                                                                                       |
+| **3** | **Understand Terraform basics**                                                                                     |
+| 3a    | Handle Terraform and provider installation and versioning                                                           |
+| 3b    | Describe plug-in based architecture                                                                                 |
+| 3c    | Demonstrate using multiple providers                                                                                |
+| 3d    | Describe how Terraform finds and fetches providers                                                                  |
+| 3e    | Explain when to use and not use provisioners and when to use local-exec or remote-exec                              |
```
Comment (lszpunar) [inline src/content/certifications/exam-faqs/terraform-associate-002.mdx:34]:
```suggestion
| 3e    | Explain when to use and not use provisioners and when to use `local-exec` or `remote-exec`                              |
```

File: src/content/certifications/exam-faqs/terraform-associate-002.mdx [hunk 2]
```diff
@@ -18,70 +18,61 @@
...
+| 7d    | Describe remote state storage mechanisms and supported standard backends                                            |
+| 7e    | Describe effect of Terraform refresh on state                                                                       |
+| 7f    | Describe backend block in configuration and best practices for partial configurations                               |
+| 7g    | Understand secret management in state files                                                                         |
+| **8** | **Read, generate, and modify configuration**                                                                        |
+| 8a    | Demonstrate use of variables and outputs                                                                            |
+| 8b    | Describe secure secret injection best practice                                                                      |
+| 8c    | Understand the use of collection and structural types                                                               |
+| 8d    | Create and differentiate resource and data configuration                                                            |
```
Comment (lszpunar) [inline src/content/certifications/exam-faqs/terraform-associate-002.mdx:67]:
```suggestion
| 8d    | Create and differentiate `resource` and `data` configuration                                                            |
```

File: src/content/certifications/exam-faqs/terraform-associate-002.mdx [hunk 3]
```diff
@@ -18,70 +18,61 @@
...
+| 8e    | Use resource addressing and resource parameters to connect resources together                                       |
+| 8f    | Use Terraform built-in functions to write configuration                                                             |
+| 8g    | Configure resource using a dynamic block                                                                            |
```
Comment (lszpunar) [inline src/content/certifications/exam-faqs/terraform-associate-002.mdx:70]:
```suggestion
| 8g    | Configure resource using a `dynamic` block                                                                            |
```

File: src/content/certifications/exam-faqs/terraform-associate-002.mdx [hunk 4]
```diff
@@ -18,70 +18,61 @@
...
+| 6e    | Execute changes to infrastructure with Terraform (`terraform apply`)                                                |
+| 6f    | Destroy Terraform managed infrastructure (`terraform destroy`)                                                      |
+| **7** | **Implement and maintain state**                                                                                    |
+| 7a    | Describe default local backend                                                                                      |
+| 7b    | Outline state locking                                                                                               |
+| 7c    | Handle backend authentication methods                                                                               |
+| 7d    | Describe remote state storage mechanisms and supported standard backends                                            |
+| 7e    | Describe effect of Terraform refresh on state                                                                       |
+| 7f    | Describe backend block in configuration and best practices for partial configurations                               |
```
Comment (lszpunar) [inline src/content/certifications/exam-faqs/terraform-associate-002.mdx:61]:
```suggestion
| 7f    | Describe `backend` block in configuration and best practices for partial configurations                               |
```

**Review by lszpunar — CHANGES_REQUESTED** [review]

**Review by lszpunar — APPROVED** [review]
Thank you thank you thank you

**Pull request merged**